lib/oli/delivery/attempts/core.ex

```diff
@@ -589,6 +589,26 @@ defmodule Oli.Delivery.Attempts.Core do
     |> Repo.preload(revision: [:activity_type])
   end
 
+  def get_latest_activity_attempt(section_id, user_id, resource_id) do
+    Repo.one(
+      from(aa in ActivityAttempt,
+        left_join: aa2 in ActivityAttempt,
+        on:
+          aa.resource_attempt_id == aa2.resource_attempt_id and
+            aa.id < aa2.id,
+        join: ra in ResourceAttempt,
+        on: ra.id == aa.resource_attempt_id,
+        join: rac in ResourceAccess,
+        on: rac.id == ra.resource_access_id,
```

Comment:
Maybe this actually should be querying to find the latest "evaluated" activity?

Reply:
Latest "evaluated" activity check added to the query